synapse/storage/databases/main/monthly_active_users.py

```diff
@@ -261,10 +264,10 @@ def _initialise_reserved_users(self, txn, threepids):
         #    run this at startup?
 
         for tp in threepids:
-            user_id = self.get_user_id_by_threepid_txn(txn, tp["medium"], tp["address"])
+            user_id = self.get_user_id_by_threepid_txn(txn, tp["medium"], tp["address"])  # type: ignore[attr-defined]
 
             if user_id:
-                is_support = self.is_support_user_txn(txn, user_id)
+                is_support = self.is_support_user_txn(txn, user_id)  # type: ignore[attr-defined]
```

Comment:
Could try making this class inherit from `RegistrationWorkerStore` to make this and the cache invalidation methods visible. It might give you MRO errors though---fine to leave as it is.

Reply:
Wow, it worked!!